Adding Changes for ibm_db_sa feedstock creating request

amukherjee28 opened the pull request:
Checklist
- [x] Adding support for ibm_db_sa package into conda-forge channel
- [x] License file is packaged (see [here](https://github.com/amukherjee28/staged-recipes/blob/master/LICENSE.txt)).
- [x] Source is from official source. https://pypi.org/project/ibm-db-sa/
- [x] Build number is 1.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://github.com/ibmdb/python-ibmdb) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://github.com/ibmdb/python-ibmdb/blob/master/README.md) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ibm_db_sa```) and found it was in an excellent condition.

amukherjee28 commented:
@conda-forge-admin @conda-forge/help-python This PR is ready to be reviewed.

xhochy reviewed (changes requested):
None of my open comments were addressed here. Please also don't open new PRs but keep pushing to existing ones.

amukherjee28 commented:
@xhochy 

The branch in pull request #14340 has some issues and hence had to raise a new pull request. I will now maintain this pull PR and will pushing changes in here.
Coming to the PR review comments following were addressed in this PR.

```
1. recipes/ibm_db_sa/meta.yaml
    - python
    - setuptools
    - libxml2
    - sqlalchemy  1.3.23
 
@xhochy xhochy on Mar 30 Member
Why this this pinned so hard?
```

The the ibm_db_sa driver supports SQLAlchemy version till 1.3.x. In case the version was not mention the sqlalchemy version 1.4.x was being installed which caused the test to fail. Hence I have pinned the version along with the package name.

```
2.recipes/ibm_db_sa/meta.yaml
  license: Apache-2.0
  license_family: Apache
  license_file: LICENSE
  summary: Python Alchemy driver for DB2 (LUW, zOS, i5) and IDS
 
@xhochy xhochy on Mar 30 Member
Should read SQLAlchemy
```

ADDRESSED THIS

```
3.recipes/ibm_db_sa/meta.yaml
  run:
    - python
    - setuptools
    - libxml2
 
@xhochy xhochy on Mar 30 Member
libxml2 should be sufficient in host
```

Removed libxml2 from run

```
4.recipes/ibm_db_sa/conda_build_config.yaml
cudnn:                                    # [linux64]
  - undefined                             # [linux64]
cuda_compiler_version:                    # [linux64]
  - None                                  # [linux64]
docker_image:                             # [linux64]
  - condaforge/linux-anvil-cos7-x86_64    # [linux64]
cdt_name:                                 # [linux64]
  - cos7                                  # [linux64]
Comment on lines +1 to +8
 
@xhochy xhochy on Mar 30 Member
This is not needed, you can add the following to conda-forge.yml on the feedstock:
```

REMOVED THE FILE

```
5.recipes/ibm_db_sa/meta.yaml
    - {{ compiler('c') }}
    - {{ compiler('cxx') }}
Comment on lines +19 to +20
 
@xhochy xhochy on Mar 30 Member
The packages doesn't seem to contain any compiled code so you probably don't need a compiler.
```

REMOVED the compiler dependency from build section

```
6.recipes/ibm_db_sa/meta.yaml
  sha256: aed72b7275ed5327501f179a0302eaf8934e084aeb3c27a7e0b68f1c83f06173

build:
  number: 1
 
@xhochy xhochy on Mar 30 Member
Should be ß
```

I have kept the build number as 1. Can you please clarify on this a bit as what the value should be.

```
7.LICENSE.txt
CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
Apache License
 
@xhochy xhochy on Mar 30 Member
Don't change the licence of the staged-recipes repository.
```

RESTORED the original content of license.txt

Please have a look at this PR and let me know.

Thanks a lot for looking into this.

xhochy commented on LICENSE.txt:
```diff
@@ -24,4 +24,4 @@ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
There is still an edit. This file shouldn't be part of the diff.

amukherjee28 replied:
I have checked again and the content of the license.txt is exactly that of license.txt in https://github.com/conda-forge/staged-recipes/blob/master/LICENSE.txt
Can you please have at a look at this once more.

xhochy replied:
There is still this diff shown in the UI. It is only a single newline character but there shouldn't be an edit here.

amukherjee28 replied:
I have replaced the file with the one in master.
Now there should not be any difference

xhochy commented on recipes/ibm_db_sa/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+build:
+  number: 1
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [linux64]
```
Removing this will fail one Linux job but spawn another. It's OK if the initial one is failing.
```suggestion
```

amukherjee28 replied:
Addressed

xhochy commented on recipes/ibm_db_sa/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+  sha256: aed72b7275ed5327501f179a0302eaf8934e084aeb3c27a7e0b68f1c83f06173
+
+build:
+  number: 1
```
```suggestion
  number: 0
```

amukherjee28 replied:
addressed

xhochy commented on recipes/ibm_db_sa/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+  run:
+    - python
+    - setuptools
+    - sqlalchemy  1.3.23
```
If it supports everything up to 1.4 (exclusive), it will better allow the full range to make it easier for users to install.
```suggestion
    - sqlalchemy  <1.4
```

amukherjee28 replied:
addressed
